src/meta/src/rpc/server.rs

```diff
@@ -171,316 +153,312 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
+
+    use super::*;
+
+    static _SLEEP_SEC: u64 = 6;
+
+    /// Start `n` meta nodes on localhost. First node will be started at `meta_port`, 2nd node on
+    /// `meta_port + 1`, ...
+    async fn _setup_n_nodes(n: u16, meta_port: u16) -> Vec<(JoinHandle<()>, WatchSender<()>)> {
+        let meta_store = Arc::new(MemStore::default());
+
+        let mut node_controllers: Vec<(JoinHandle<()>, WatchSender<()>)> = vec![];
+        for i in 0..n {
+            let addr = format!("http://127.0.0.1:{}", meta_port + i);
+
+            let info = AddressInfo {
+                addr,
+                listen_addr: SocketAddr::new(
+                    IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+                    meta_port + i,
+                ),
+                ..AddressInfo::default()
+            };
+            node_controllers.push(
+                rpc_serve_with_store(
+                    meta_store.clone(),
+                    info,
+                    Duration::from_secs(4),
+                    2,
+                    MetaOpts::test(false),
+                )
+                .await
+                .unwrap_or_else(|e| panic!("Meta node{} failed in setup. Err: {}", i, e)),
+            );
+        }
+        sleep(Duration::from_secs(_SLEEP_SEC)).await;
+        node_controllers
+    }
 
-    let compactor_manager = Arc::new(
-        hummock::CompactorManager::with_meta(env.clone(), max_heartbeat_interval.as_secs())
-            .await
-            .unwrap(),
-    );
-
-    let hummock_manager = hummock::HummockManager::new(
-        env.clone(),
-        cluster_manager.clone(),
-        meta_metrics.clone(),
-        compactor_manager.clone(),
-    )
-    .await
-    .unwrap();
-
-    #[cfg(not(madsim))]
-    if let Some(ref dashboard_addr) = address_info.dashboard_addr {
-        let dashboard_service = crate::dashboard::DashboardService {
-            dashboard_addr: *dashboard_addr,
-            cluster_manager: cluster_manager.clone(),
-            fragment_manager: fragment_manager.clone(),
-            meta_store: env.meta_store_ref(),
-            prometheus_endpoint: prometheus_endpoint.clone(),
-            prometheus_client: prometheus_endpoint.as_ref().map(|x| {
-                use std::str::FromStr;
-                prometheus_http_query::Client::from_str(x).unwrap()
-            }),
-        };
-        // TODO: join dashboard service back to local thread.
-        tokio::spawn(dashboard_service.serve(address_info.ui_path));
+    /// Check for `number_of_nodes` meta leader nodes, starting at `meta_port`, `meta_port + 1`, ...
+    /// Simulates `number_of_nodes` compute nodes, starting at `meta_port`, `meta_port + 1`, ...
+    ///
+    /// ## Returns
+    /// Number of nodes which currently are leaders. Number is not snapshoted. If there is a
+    /// leader failover in process, you may get an incorrect result
+    async fn _number_of_leaders(number_of_nodes: u16, meta_port: u16, host_port: u16) -> u16 {
```

Comment:
```suggestion
    async fn number_of_leaders(number_of_nodes: u16, meta_port: u16, host_port: u16) -> u16 {
```
Check the number is okay for now. We can check more when meta member service get ready.

Reply:
Will use `#[cfg(test)]`. Same as above.

Reply:
I agree. For should for example test if the reported current leader from the discovery service is always reported to be the same leader. I wrote down test ideas in https://github.com/risingwavelabs/risingwave/issues/6884. Feel free to add your own suggestions